enhancements/network/ingress-node/ingress-node-firewall-with-ebpf.md

```diff
@@ -0,0 +1,284 @@
+---
+title: ingress-node-firewall-with-ebpf
+
+authors:
+  - "@msherif1234"
+
+reviewers:
+  - "@danwinship"
+  - "@dave-tucker"
+  - "@dcbw"
+  - "@trozet"
+
+approvers:
+  - "@danwinship"
+  - "@knobunc"
+
+api-approvers:
+  - "@danwinship"
+
+creation-date: 2022-06-15
+
+last-updated: 2022-07-19
+
+status: implementable
+
+tracking-link:
+  - https://issues.redhat.com/browse/SDN-2781
+---
+
+# Ingress node Firewall using eBPF
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+OCP customers would like to secure OCP's nodes from external attacks by configuring ingress firewall policies.
+eBPF provides a very flexible filtering mechanism to allow early detection and filtering.
+eXpress Data Path (XDP) is a feature in the Linux kernel which allows users to execute a user-supplied eBPF program when a packet is received on a network interface (NIC). 
+This program is executed in the NIC’s device driver, right after the interrupt processing and before the network stack. 
+The goal of XDP is to offer comparable performance to kernel bypass solutions while working with the existing kernel networking stack.
+A new cluster-scoped object `IngressNodeFirewall` will be used to configure node ingress firewall match rules and per rule statistics will be avaliable.
+Admin can provision multiple instances of this object while ensuring certain execution order.
+For the best performance XDP programs need to attach to interfaces using the following models:
```

Comment:
FWIW, the kernel network team has (belatedly) suggested that we shouldn't even use XDP if the NIC doesn't support native XDP...

Reply:
the prob with NIC doesn't support XDP performance will just degrade become as slow as doing skb kernel processing so going from MPPS to KPPS but that is expected and explainable